docs/DeveloperGuide.md

```diff
@@ -392,6 +392,52 @@ Sequence diagram for the deletion of a loan:
       principles
       ,and makes the code harder to maintain.
 
+### Loan view command - Wang Junwu
+
+#### Implementation
+
+The `ViewLoanCommand` class handles the viewing of all loans attached to a contact, and executes the command after the
+input is parsed and transformed into an appropriate format.
+The parsing of the command is done by the `ViewLoanCommandParser` class,
+which is responsible for parsing the user input.
+
+The `ViewLoanCommand` class is instantiated in the `ViewLoanCommandParser` class, while the
+`ViewLoanCommandParser` is instantiated in the `AddressBookParser` class. Both classes are instantiated when the user
+enters a `viewloan` command, which needs to be of the format `viewloan INDEX`
+where INDEX is the index of the person whose loans are to be viewed, a positive whole number.
+
+The `ViewLoanCommand` class contains the following fields which can prove to be useful for the user:
+* `targetIndex`: the index of the person whose loans are to be viewed
+* Some string fields that are displayed to the user under different scenarios.
+
+The `ViewLoanCommandParser` class does not contain any fields.
+
+Sequence diagram for the viewing of loans:
+![ViewLoanSequenceDiagram](images/ViewLoanSequenceDiagram.png)
+
+#### Design considerations:
+
+##### Aspect: How the loan list is accessed:
+
+* **Alternative 1 (current choice):** The `ViewLoanCommand.execute()` method sequentially obtains the person list,
```

Comment:
Hmm, we can consider implementing Alternative 2 in the future to make it SLAP-compliant. Thanks for pointing this out!